### ISSUE TITLE
SP not working with units

### ISSUE THREAD

**Issue by pgkirsch** [convexopt/gpkit#421]
```python
from gpkit.shortcuts import *
from gpkit import SignomialsEnabled

x = Variable('x', 'm')
y = Variable('y', 'm')
z = Variable('z', 2, 'm')
xmin = Variable('xmin', 0.1, 'm')

with SignomialsEnabled():
    m = Model(x, [x + y >= z,
                  x >= xmin])

m.localsolve()
```

gives an error
```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/Users/philippekirschen/Documents/MIT/Research/GPkit/gpkit/gpkit/tests/test_SPunits.py in <module>()
     11                   x >= xmin])
     12 
---> 13 m.localsolve()

/Users/philippekirschen/Documents/MIT/Research/GPkit/gpkit/gpkit/model.pyc in localsolve(self, solver, verbosity, skipfailures, *args, **kwargs)
    245             with SignomialsEnabled():
    246                 return self._solve("sp", solver, verbosity, skipfailures,
--> 247                                    *args, **kwargs)
    248         except ValueError as err:
    249             if err.message == ("SignomialPrograms must contain at least one"

/Users/philippekirschen/Documents/MIT/Research/GPkit/gpkit/gpkit/model.pyc in _solve(self, programType, solver, verbosity, skipfailures, *args, **kwargs)
    344         else:
    345             signomials, beforesubs.smaps = simplify_and_mmap(signomials,
--> 346                                                            constants)
    347             # NOTE: SIDE EFFECTS
    348             self.program, solvefn = form_program(programType, signomials,

/Users/philippekirschen/Documents/MIT/Research/GPkit/gpkit/gpkit/model.pyc in simplify_and_mmap(constraints, subs)
    572             exps, cs, smap = simplify_exps_and_cs(exps, cs, return_map=True)
    573             if s.any_nonpositive_cs:
--> 574                 negative_c_count = (cs <= 0).sum()
    575                 if negative_c_count == 0:
    576                     raise RuntimeWarning("""Infeasible SignomialConstraint.

/Users/philippekirschen/Documents/MIT/Research/GPkit/gpkit/gpkit/__init__.pyc in newfn(self, other)
    136                 return NotImplemented
    137             else:
--> 138                 return getattr(self, fallback)(other)
    139         return newfn
    140 

/Users/philippekirschen/anaconda/lib/python2.7/site-packages/pint/quantity.pyc in <lambda>(self, other)
    816 
    817     __lt__ = lambda self, other: self.compare(other, op=operator.lt)
--> 818     __le__ = lambda self, other: self.compare(other, op=operator.le)
    819     __ge__ = lambda self, other: self.compare(other, op=operator.ge)
    820     __gt__ = lambda self, other: self.compare(other, op=operator.gt)

/Users/philippekirschen/anaconda/lib/python2.7/site-packages/pint/quantity.pyc in compare(self, other, op)
    805                 return op(self._convert_magnitude_not_inplace(UnitsContainer()), other)
    806             else:
--> 807                 raise ValueError('Cannot compare Quantity and {0}'.format(type(other)))
    808 
    809         if self.units == other.units:

ValueError: Cannot compare Quantity and <type 'int'>
```

**Comment by bqpd** [comment]
Oh, weird. I'll take a look.

**Comment by bqpd** [comment]
Fixed in #430 (was missing a mag)

**Comment by whoburg** [comment]
Let's let the PR close this?
